### PR TITLE
Issue #2833: capture and pass in editor args to edit cli command

### PR DESCRIPTION
**Pull request by cabrinha**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to the README.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

File: util/cli/cli.go [hunk 1]
```diff
@@ -204,16 +204,21 @@ func setComments(input []byte, comments string) []byte {
...
 }
 
+// InteractiveEdit launches an interactive editor
 func InteractiveEdit(filePattern string, data []byte, save func(input []byte) error) {
 	editor := os.Getenv(editorEnv)
 	if editor == "" {
 		editor = defaultEditor
 	}
+	var editorArgs []string
+	if strings.Contains(editor, " ") {
```
Comment (alexec) [inline util/cli/cli.go]: I don't think you need the guard condition.

Comment (alexec) [reply util/cli/cli.go]: This does need tests mind you.

Comment (cabrinha) [reply util/cli/cli.go]: yeah so it looks like the circle tests are failing here: https://app.circleci.com/jobs/github/argoproj/argo-cd/10188

And when I try to `make cli` locally, I get a similar error. I'm not sure how to resolve it. Also pinged you on slack.

**Comment by cabrinha** [comment]
Somehow I was able to screw up this PR by merging in other changes. Closing for now.

**Pull request closed without merging**